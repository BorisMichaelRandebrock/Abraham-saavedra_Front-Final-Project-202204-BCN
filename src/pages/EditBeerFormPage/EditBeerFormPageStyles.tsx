import styled from "styled-components";

const EditBeerFormPageStyles = styled.section`
  margin: 100px auto;
  border: 1px solid #b6b6b6;
  width: 220px;
  height: 550px;
  border-radius: 20px;
  box-shadow: 10px 10px 7px #b6b6b6;
`;

export default EditBeerFormPageStyles;
