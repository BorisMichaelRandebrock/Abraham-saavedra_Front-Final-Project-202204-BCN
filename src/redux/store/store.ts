import { configureStore } from "@reduxjs/toolkit";
import beerReducer from "../features/beerSlice";
import uiReducer from "../features/uiSlice";
import userReducer from "../features/userSlice";

export const store = configureStore({
  reducer: {
    user: userReducer,
    beer: beerReducer,
    ui: uiReducer,
  },
});

export type RootState = ReturnType<typeof store.getState>;

export type AppDispatch = typeof store.dispatch;
