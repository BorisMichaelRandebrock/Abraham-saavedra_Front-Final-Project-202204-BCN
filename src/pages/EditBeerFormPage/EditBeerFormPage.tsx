import CreateEditBeerForm from "../../components/CreateEditBeerForm/CreateEditBeerForm";
import EditBeerFormPageStyles from "./EditBeerFormPageStyles";

const EditBeerFormPage = (): JSX.Element => {
  return (
    <EditBeerFormPageStyles>
      <CreateEditBeerForm />
    </EditBeerFormPageStyles>
  );
};

export default EditBeerFormPage;
