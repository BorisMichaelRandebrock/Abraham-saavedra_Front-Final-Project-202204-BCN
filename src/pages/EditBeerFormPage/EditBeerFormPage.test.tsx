import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { store } from "../../redux/store/store";
import EditBeerFormPage from "./EditBeerFormPage";

describe("Given a EditBeerFormPage component", () => {
  describe("When it's rendered", () => {
    test("Then it sould show button to editBeer", () => {
      const buttonName = "Editar cerveza";

      render(
        <BrowserRouter>
          <Provider store={store}>
            <EditBeerFormPage />
          </Provider>
        </BrowserRouter>
      );
      const button = screen.getByRole("button", {
        name: buttonName,
      });

      expect(button).toBeInTheDocument();
    });
  });
});
