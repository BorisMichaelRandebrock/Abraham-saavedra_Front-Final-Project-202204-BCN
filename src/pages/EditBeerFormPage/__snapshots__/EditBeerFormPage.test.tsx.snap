// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a EditBeerFormPage component When it's rendered Then it sould show button to editBeer 1`] = `
<section
  className="sc-gsnTZi eGTtUg"
/>
`;
